ml_garden/core/steps/fit_model.py

```diff
@@ -124,7 +131,7 @@ def _create_study(self) -> optuna.Study:
         return study
 
     @staticmethod
-    def _calculate_error(y_true, y_pred, metric):
+    def _calculate_error(y_true: np.ndarray, y_pred: np.ndarray, metric: str, task: str) -> float:
```

Comment:
Not sure about the library, but maybe you could support `numpy.typing.ArrayLike`s instead of just `np.ndarray`s, since most numpy functions actually support being called with other things (like `list`s).

This is very low priority I'd say, but it could help if passing a `pd.Series`.